test/viewer.js

```diff
@@ -0,0 +1,50 @@
+const net = require('net');
+
+const Logger = require('../lib/Logger');
+const { startServer } = require('../lib/viewer.js');
+
+describe('WebSocket server', function () {
+  it('should not crash when an error is emitted on the websocket', function (done) {
+    const bundleStats = {
+      assets: [{ name: 'bundle.js', chunks: [0] }]
+    };
+
+    const options = {
+      openBrowser: false,
+      logger: new Logger('silent'),
+      port: 0
+    };
+
+    startServer(bundleStats, options).then(function ({ http: server }) {
+      const socket = net.createConnection(server.address().port, function () {
+        socket.write([
+          'GET / HTTP/1.1',
+          'Host: localhost',
+          'Upgrade: websocket',
+          'Connection: Upgrade',
+          'Sec-WebSocket-Key: Ix02x1E8kmC9ie01/dGCuw==',
+          'Sec-WebSocket-Version: 13',
+          '\r\n'
+        ].join('\r\n'));
+      });
+
+      socket.on('data', function (chunk) {
+        const expected = Buffer.from([
+          'HTTP/1.1 101 Switching Protocols',
+          'Upgrade: websocket',
+          'Connection: Upgrade',
+          'Sec-WebSocket-Accept: XYMa8DJmsxHNEsGJZwEZT7XX1Dk=',
+          '\r\n'
+        ].join('\r\n'));
+
+        expect(chunk.equals(expected)).to.be.true;
+
+        // Send a WebSocket frame with a reserved opcode: 5
+        socket.write(Buffer.from([0x85, 0x00]));
```

Comment:
Is this the part where an error is triggered? I'm not that well versed in how websocket protocol works.

If writing a reserved opcode indeed causes the error to be triggered, maybe the comment could be amended to say that explicitly? For example, something like this:

```js
// Send a WebSocket frame with a reserved opcode (5) to trigger
// an error to be emitted on the server
```

Reply:
Yes this how the error is triggered on the server, I will update the comment.